src/public-modules/Bounty/sagas.js

```diff
@@ -493,6 +491,84 @@ export function* increasePayout(action) {
   }
 }
 
+export function* contribute(action) {
+  const { id, value, paysTokens, decimals, tokenContract } = action;
+  const userAddress = yield select(addressSelector);
+  yield put(setPendingWalletConfirm());
+  let addedBalance;
+  if (paysTokens) {
+    addedBalance = calculateDecimals(
+      BigNumber(value, 10).toString(),
+      parseInt(decimals)
+    );
+  } else {
+    const { web3 } = yield call(getWeb3Client);
+    addedBalance = web3.utils.toWei(BigNumber(value, 10).toString(), 'ether');
+  }
+  try {
+    const { standardBounties } = yield call(getContractClient);
+    let txHash;
+    if (paysTokens) {
+      const { tokenContract: tokenContractClient } = yield call(
+        getTokenClient,
+        tokenContract
+      );
+      const network = yield select(networkSelector);
+      yield call(
+        promisifyContractCall(tokenContractClient.approve, {
+          from: userAddress
+        }),
+        config[network].standardBountiesAddress,
+        addedBalance
+      );
+      yield call(delay, 2000);
```

Comment:
Referencing issue:
https://github.com/MetaMask/metamask-extension/issues/3425

Batching didn't work previously, and tagging them one after another also didn't work.

Reply:
@danfinlay FYI - batching not properly working with the approve then call function.

Reply:
Putting it back to delay.